crates/wast/src/resolve/names.rs

```diff
@@ -1730,7 +1809,14 @@ impl<'a, 'b> ExprResolver<'a, 'b> {
             }
 
             LocalSet(i) | LocalGet(i) | LocalTee(i) => {
-                self.locals.resolve(i, "local")?;
+                for (depth, scope) in self.scopes.iter().enumerate().rev() {
+                    if let Err(e) = scope.resolve(i, "local") {
+                        if depth == 0 {
+                            return Err(e);
+                        }
+                    }
```

Comment:
Should this `break` once the local is resolved?

Reply:
Yup, nice catch. Impressed that it worked without that.